src/main/scala/cognite/spark/v1/DefaultSource.scala

```diff
@@ -274,9 +275,13 @@ class DefaultSource
       // If the number of partitions is reasonable, we avoid the data shuffling
       val (dataRepartitioned, numberOfPartitions) =
         if (originalNumberOfPartitions > 50 && originalNumberOfPartitions > idealNumberOfPartitions) {
-          (data.repartition(idealNumberOfPartitions), idealNumberOfPartitions)
+          val partitionsAfterShrink =
+            (idealNumberOfPartitions * (100 - config.partitionShrinkPercentage) / 100).toInt
+          (data.repartition(partitionsAfterShrink), partitionsAfterShrink)
         } else {
-          (data, originalNumberOfPartitions)
+          val partitionsAfterShrink =
+            (originalNumberOfPartitions * (100 - config.partitionShrinkPercentage) / 100).toInt
+          (data.repartition(partitionsAfterShrink), partitionsAfterShrink)
```

Comment:
I think you should avoid calling repartition unless it's needed. AFAIK Spark will always do the data shuffling, even when the number of partitions remains exacrly the same.

Anyway, why do you need this option? Can't you just decrease number of `sparkPartitions`? (by increasing parallelismPerPartition, iirc)